src/ui-test/tests/basic_mode_test.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License", destination); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+import * as consts from '../utils/uiTestConstants';
+import { EditorView, SideBarView, VSBrowser, WebDriver } from 'vscode-extension-tester';
+import { prepareEmptyTestFolder } from '../utils/utils';
+import { basicIntegration } from '../../IntegrationConstants';
+import {
+    cleanOutputView,
+    viewHasItem,
+    contextMenuItemClick,
+    webViewHasTextInWebElement
+} from '../utils/waitConditions';
+import {
+    textDoesNotContainAsci,
+    inputBoxQuickPickOrSet,
+    findSectionItem,
+    DoNextTest
+} from '../utils/utils';
+import { assert } from 'chai';
+
+export function basicModeWithLogsTest(extension: string, language: string, doNextTest: DoNextTest) {
+
+    describe(`Basic mode test with the logs`, function () {
+
+        let driver: WebDriver;
+
+        const [, , , updatedLogMessage] = consts.prepareCodeLogMessages(extension, language);
+
+        before(async function () {
+            driver = VSBrowser.instance.driver;
+        });
+
+        after(async function () {
+            this.timeout(60000);
+            await new EditorView().closeAllEditors();
+            const item = await findSectionItem(consts.extensionName, consts.integrationFileName.toLowerCase());
+            if (item !== undefined) {
+                await driver.wait(() => { return contextMenuItemClick(item, consts.integrationRemove); });
+                const content = new SideBarView().getContent();
+                await driver.wait(async () => {
+                    return viewHasItem(content, consts.extensionName, consts.integrationFileName.toLowerCase())
```

Comment:
It sounds a simple enough single statement for me
`return !(await viewHasItem(content, consts.extensionName, consts.integrationFileName.toLowerCase()));`
Which can even be written on several lines if it is the length of lien which is too long:
```
return !(await viewHasItem(content,
                                           consts.extensionName,
                                           consts.integrationFileName.toLowerCase()));`
```

Reply:
Agree with @apupier, think we can close that request cause it is duplicated

Reply:
I dunno, I have to parse a lot to understand that:

```
!(await viewHasItem(content,
                                           consts.extensionName,
                                           consts.integrationFileName.toLowerCase()));`
```
                                         
means:

```
not ( content->viewHasItem(extensionName, integrationFileName)) //And why is the integration File needed...?
```

instead of having something like:

```
return viewHasNotIntegrationFile(content, integrationLabel)
```

I feel like an overall issue with the test suites is that they are really hard to scan. There are a lot of callbacks which are not redefined as functions with good names, making reading them hard, specially when I was starting.  I thought this was a good place to change it and make it easier to read, but if you think this is clear enough we can go ahead.

Reply:
Thanks, @joshiraez, I got your idea, due the fact I used the same code block three times (and will use it more), I added specific function -> `sidebarIntegrationRemove` (to remove integration through the sidebar). 

Yes, it was my first step in UI tests with extester, so I fully understand you in  `And without any prior knowledge of extester or the library I have to try them myself or start delving deeper into the code to understand it what it wanted to do` :-), during the next PR I will try to add also generic tests for repetitive operations.

I hope it will be easy to create tests with steps like: create integration file,  check file, update file, check integration started, check output/log, check integration in sidebar and etc with ready generic tests.

Reply:
Yes, that'd help, definitely

I tried creating something like a generic test object with generic operations after an Aurelien suggestion on my own PR but didnt go well :) 

Ill be happy to help with the template tests if I can helo in some way, or keeo adding to them in the future ^^